gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -434,6 +434,97 @@ gst_tensor_filter_finalize (GObject * object)
   G_OBJECT_CLASS (parent_class)->finalize (object);
 }
 
+/**
+ * @brief Parse the string of model
+ * @param info tensors info structure
+ * @param model_file the prediction model path
+ * @param model_file_sub the initialize model path
+ * @return number of parsed model path
+ */
+static guint
+gst_tensors_parse_modelpaths_string (GstTensorFilterProperties * prop,
+    const gchar * model_files)
+{
+  gchar **models;
+  gchar **model_0;
+  gchar **model_1;
+  guint num_models = 0;
+  guint num_model_0 = 0;
+  guint num_model_1 = 0;
+
+  g_return_val_if_fail (prop != NULL, 0);
+
+  if (model_files) {
+
+    models = g_strsplit_set (model_files, ",", -1);
+
+    num_models = g_strv_length (models);
+
+    if (num_models == 1) {
+      prop->model_file = g_strdup (models[0]);
+      return num_models;
+    }
+
+    else if (num_models == 2) {
+      model_0 = g_strsplit_set (models[0], "=", -1);
+      model_1 = g_strsplit_set (models[1], "=", -1);
+
+      num_model_0 = g_strv_length (model_0);
+      num_model_1 = g_strv_length (model_1);
+
+      if (num_model_0 == 1 && num_model_1 == 1) {
+        prop->model_file_sub = g_strdup (model_0[0]);
+        prop->model_file = g_strdup (model_1[0]);
+      }
+
+      else if (g_strcmp0 (model_0[0], "init") == 0 ||
+          g_strcmp0 (model_0[0], "Init") == 0) {
+
+        prop->model_file_sub = g_strdup (model_0[1]);
+
+        if (num_model_1 == 2)
+          prop->model_file = g_strdup (model_1[1]);
+        else
+          prop->model_file = g_strdup (model_1[0]);
+      }
+
+      else if (g_strcmp0 (model_0[0], "pred") == 0 ||
+          g_strcmp0 (model_0[0], "Pred") == 0) {
+
+        prop->model_file = g_strdup (model_0[1]);
+
+        if (num_model_1 == 2)
+          prop->model_file_sub = g_strdup (model_1[1]);
+        else
+          prop->model_file_sub = g_strdup (model_1[0]);
+      }
+
+      else if (g_strcmp0 (model_1[0], "init") == 0 ||
+          g_strcmp0 (model_1[0], "Init") == 0) {
+
+        prop->model_file_sub = g_strdup (model_1[1]);
+
+        if (num_model_0 == 2)
+          prop->model_file = g_strdup (model_0[1]);
+        else
+          prop->model_file = g_strdup (model_0[0]);
+      }
+
+      else if (g_strcmp0 (model_1[0], "pred") == 0 ||
+          g_strcmp0 (model_1[0], "Pred") == 0) {
+
+        prop->model_file = g_strdup (model_1[1]);
+
+        if (num_model_0 == 2)
+          prop->model_file_sub = g_strdup (model_0[1]);
+        else
+          prop->model_file_sub = g_strdup (model_0[0]);
+      }
+    }
```

Comment:
I think
g_strfreev(model_0); g_strfreev(model_1);  g_strfreev(models);

Reply:
it's applied